x-pack/plugin/fleet/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/fleet/20_wait_for_checkpoints.yml

```diff
@@ -0,0 +1,83 @@
+setup:
+  - do:
+      indices.create:
+        index: test-timeout
+        body:
+          settings:
+            number_of_shards: "2"
+
+  - do:
+      indices.create:
+        index: test-after-refresh
+        body:
+          settings:
+            number_of_shards: "1"
+
+  - do:
+      indices.put_alias:
+        index: test-after-refresh
+        name: test-alias
+
+  - do:
+      index:
+        index: test-timeout
+        body: { }
+
+  - do:
+      index:
+        index:  test-after-refresh
+        body:   { }
+
+  - do:
+      index:
+        index: test-after-refresh
+        body: { }
+
+  - do:
+      indices.refresh: {}
+
+---
+"Execute successful after refresh search":
+  - do:
+      fleet.search:
+        index: "test-after-refresh"
+        allow_partial_search_results: false
+        wait_for_checkpoints: [1]
+        body: { query: { match_all: {} } }
+
+  - match: { _shards.successful: 1 }
+  - match: { hits.total.value: 2 }
+
+---
+"Cannot use alias":
+  - do:
+      catch: bad_request
+      fleet.search:
+        index: "test-alias"
+        allow_partial_search_results: false
+        wait_for_checkpoints: [1]
+        body: { query: { match_all: {} } }
+
+---
+"After refresh timeout":
```

Comment:
Is it really a timeout test ? It looks like the check will fail because the checkpoint is after the current engine checkpoint.